Modules/Core/SpatialObjects/include/itkSpatialObjectProperty.h

```diff
@@ -91,18 +91,12 @@ class ITKSpatialObjects_EXPORT SpatialObjectProperty
   GetAlpha() const;
 
   void
-  SetName(const std::string & name)
+  SetName(std::string name)
```

Comment:
This change might cause a performance penalty when the string is large. What about using C++17 `std::string_view`? string_view is designed to be passed by-value, and it supports both `std::string` and `const char*` arguments efficiently (avoiding dynamic memory allocation).

Reply:
Agreed - but I considered the issue of having it throw a warning message in python (see the commit message examples) versus the chance that the "name" of an object would be a long string that could cause copy and/or memory issues, and I thought getting rid of the python message was more likely/prevalent/important.    Perhaps this could be considered a necessary evil?

We could also start changing things to string_view throughout the code, but perhaps that is better as a separate pull request rather that did that on a large portion of ITK, rather than hiding/embedding it as a change one function at a time?

Reply:
ITK does have `const std::string &` parameters in many other places. Are all of these problematic for SWIG?

Reply:
@thewtex Does nonobind have the same problem with returning string references?

Reply:
Great question!!!   

@blowekamp  - wanted flag this for you, since the message trail is getting "long".

Reply:
nanobind may have improved behavior, but I do not know the case for `const std::string &`.